src/librustc/hir/lowering.rs

```diff
@@ -1823,10 +1824,22 @@ impl<'a> LoweringContext<'a> {
                         (hir::GenericArgs::none(), true)
                     }
                     ParenthesizedGenericArgs::Err => {
-                        struct_span_err!(self.sess, data.span, E0214, "{}", msg)
-                            .span_label(data.span, "only traits may use parentheses")
-                            .emit();
-                        (hir::GenericArgs::none(), true)
+                        let mut err = struct_span_err!(self.sess, data.span, E0214, "{}", msg);
+                        err.span_label(data.span, "only traits may use parentheses");
```

Comment:
No time to dig into it now, but I'm mildly curious about the history of this rule. Offhand, I don't seem to _recall_ seeing traits use parentheses for type argum—oh, never mind, I just read `--explain 214`, and it points out that it's for `Fn(T)` types.

This does seem sufficiently remote from the crux of the error at hand that we might want to reword the top-line error message to emphasize "parameters use brackets, generally" rather than highlighting the unusual case of traits? I don't have a crisp alternative wording to propose at the moment, though.